plugins/flashrom/fu-plugin-flashrom.c

```diff
@@ -0,0 +1,220 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+ *
+ * Licensed under the GNU General Public License Version 2
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "config.h"
+
+#include <string.h>
+
+#include "fu-plugin.h"
+#include "fu-plugin-vfuncs.h"
+
+/**
+ * FU_QUIRKS_FLASHROM_HWID:
+ * @key: a GUID, e.g. `HwId=a0ce5085-2dea-5086-ae72-45810a186ad0`
+ * @value: a device ID, e.g. `librem15v3`
+ *
+ * When the system motherboard can be updated using flashrom, this quirk maps
+ * a hardware ID to a device ID.
+ *
+ * The HWID can be found using the `fwupdmgr hwids` command.
+ *
+ * Since: 1.1.0
+ */
+#define	FU_QUIRKS_FLASHROM_DEVICE_ID		"DeviceId"
+
+struct FuPluginData {
+	gchar			*flashrom_fn;
+};
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
+}
+
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	g_free (data->flashrom_fn);
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	GPtrArray *hwids;
+
+	/* we need flashrom from the host system */
+	data->flashrom_fn = g_find_program_in_path ("flashrom");
+	if (data->flashrom_fn == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "flashrom is not installed");
+		return FALSE;
```

Comment:
Since we've been switching over to marking devices not updatable and setting `UpdateError`, I think this is a good candidate to do the same rather than fail plugin startup.

Reply:
Agree. I'm still not sure if I want to merge this plugin or wait for libflashrom to happen. It doesn't seem like the latter is coming any time soon.

Reply:
I say update for that change merge it now, and if `libflashrom` happens, clean up later, bump up the dependency.